### PR TITLE
[Publish] Dialog - DGST-36

### PR DESCRIPTION
**Pull request by joesnellpdx**
**Publish Dialog**

-- Currently WIP / Not ready for Review --
1. Content approved here: https://jdrf.atlassian.net/browse/DGST-35
2. Update content and update this MR as ready for review

To test / review:
- Pull branch
- Go to `/doc-site` in terminal
- run `npm run-link-local-library`
- run `gulp`
- ` go to `/components/dialog.html` and test/review

### CODE REVIEW

File: doc-site/docs/components/dialog.njk [hunk 1]
```diff
@@ -0,0 +1,721 @@
+{% extends 'templates/doc.njk' %}
+{% block page_shell_content_intro %}
+    {% filter markdown %}
+        <h1>Dialog</h1>
+
+        <p class="spirit-subhead">Dialogs focus the user’s attention on a single task or piece of information and often require decisions. They appear over the interface and block further interactions.</p>
+        <p>Dialogs are disruptive intentionally and should be used thoughtfully and sparingly depending on the informational context and/or need.</p>
```
Comment (patrickmckowen) [inline doc-site/docs/components/dialog.njk:7]:
```suggestion
        <p>Dialogs are disruptive intentionally and should be used thoughtfully and sparingly depending.</p>
```

File: doc-site/docs/components/dialog.njk [hunk 2]
```diff
@@ -0,0 +1,721 @@
...
+        <p>Dialogs depend on Javascript. We are leveraging [A11y Dialog](https://edenspiekermann.github.io/a11y-dialog/) within the Spirit design system to run the dialogs. See [Guidelines - JavaScript](#javascript) for information javascript usage and the necessary dependencies.</p>
```
Comment (patrickmckowen) [inline doc-site/docs/components/dialog.njk:8]:
```suggestion
        <p>Dialogs depend on Javascript. We are leveraging [A11y Dialog](https://edenspiekermann.github.io/a11y-dialog/) within the Spirit design system to run the dialogs. See [JavaScript Guidelines](#javascript) for information on usage and dependencies.</p>
```

File: doc-site/docs/components/dialog.njk [hunk 3]
```diff
@@ -0,0 +1,721 @@
...
+        * Dismiss dialog by clicking the close button, pressing the escape key, or clicking outside the dialog.
+        * If a title is used, it is recommended to give the title a unique ID and add that ID to the `.spirit-dialog__content` container as `aria-labelledby=”title-id”` to enhance accessibility UX.
+        * The title will be truncated to one line (unless dialog is `.spirit-dialog__content--alert` - see [alert](#alert) dialog below).
+        * Multiple [Icon Only](/components/buttons.html#icon-only) button action items can be added to the `.spirit-dialog__header-icons container`. The close icon should always be the right most icon.
+
+        ## Modal
+
+        A modal dialog requires the user to confirm or cancel an action before it can be closed.
+
+        <img src="/images/dialog/dialog_modal.png" alt="Modal Dialog Example" />
```
Comment (patrickmckowen) [inline doc-site/docs/components/dialog.njk:143]: Please update image so it is 100% wide, same width as the main content.

File: doc-site/docs/components/dialog.njk [hunk 4]
```diff
@@ -0,0 +1,721 @@
...
+    {% endfilter %}
+
+    {% set dialog_modal %}
+
+        {% set dialog_id="doc-sample-modal" %}
+        {% call library.dialog(dialog_id=dialog_id, size="fullscreen", size_break="small", modal=true, has_title=true, has_close=true) %}
+            {{ library.dialog_header(title_text="Edit Content", dialog_id=dialog_id, has_header_close=false) }}
```
Comment (patrickmckowen) [inline doc-site/docs/components/dialog.njk:150]:
```suggestion
            {{ library.dialog_header(title_text="Edit Contact", dialog_id=dialog_id, has_header_close=false) }}
```

File: doc-site/docs/components/dialog.njk [hunk 5]
```diff
@@ -0,0 +1,721 @@
...
+                }
+            ],
+            code_snippet_source=(dialog_modal)
+        )
+    }}
+
+    {% filter markdown %}
+        * The dialog is "locked" open and cannot be dismissed by pressing the `escape` key, or clicking outside the dialog. This is set by changing `role="dialog"` in the `.spirit-dialog__content` element to  `role="alertdialog"`.
+        * No close [Icon Only](/components/buttons.html#icon-only) button is added to the header or at the top of the dialog.
+        * Footer is required with buttons that trigger the close of the dialog or an explicit action determined by the event. These actions require custom coding. See [Guidelines - JavaScript](#javascript) for further guidance.
```
Comment (patrickmckowen) [inline doc-site/docs/components/dialog.njk:175]:
```suggestion
        * Footer is required with buttons that trigger the close of the dialog or an explicit action determined by the event. These actions require custom coding. See [JavaScript Guidelines](#javascript) for further guidance.
```

File: doc-site/docs/components/dialog.njk [hunk 6]
```diff
@@ -0,0 +1,721 @@
...
+    {% endfilter %}
+
+    {% set dialog_alert %}
+
+        {% set dialog_id="doc-sample-alert" %}
+        {% call library.dialog(dialog_id=dialog_id, size="fullscreen", size_break="small", alert=true, has_title=true, has_close=true) %}
+            {{ library.dialog_header(title_text="Discard Changes?", dialog_id=dialog_id, has_header_close=false) }}
+            {% call library.dialog_content() %}
+                <div class="spirit-long-form-text">
+                    <p>Your pohoto has changes that have not been saved. Do you want to discard changes?</p>
```
Comment (patrickmckowen) [inline doc-site/docs/components/dialog.njk:192]:
```suggestion
                    <p>Your photo has changes that have not been saved. Do you want to discard changes?</p>
```

File: doc-site/docs/components/dialog.njk (hunk 3 above)
Comment (joesnellpdx) [reply doc-site/docs/components/dialog.njk:143]: @patrickmckowen  
I need an image for this one. 
The only image I have is 570px wide.

**Pull request merged**